src/system/mod.rs

```diff
@@ -116,11 +116,16 @@ impl FileCloser {
 
 fn close_range(min_fd: c_uint, max_fd: c_uint) -> io::Result<()> {
     if min_fd <= max_fd {
-        // SAFETY: this function is safe to call:
-        // - any errors while closing a specific fd will be effectively ignored
-        // - if the provided range or flags are invalid, that will be reported
-        //   as an error but will not cause undefined behaviour
-        cerr(unsafe { libc::close_range(min_fd, max_fd, 0) })?;
+        if cfg!(all(target_os = "linux", target_env = "musl")) {
```

Comment:
You will have to use `#[cfg]`. The code as written will still try to reference `libc::close_range` even on musl. `if cfg!()` only makes it unreachable at runtime.

Reply:
🤪 I thought so too but I have clear memories of this compiling. But now that you pointed out that it shouldn't, it doesn;'t.